spec/requests/articles/articles_show_spec.rb

```diff
@@ -15,31 +17,64 @@
       expect(response).to have_http_status(:ok)
     end
 
-    it "renders the proper title" do
-      expect(response.body).to include CGI.escapeHTML(article.title)
-    end
-
-    it "renders the proper published at date" do
-      expect(response.body).to include CGI.escapeHTML(article.readable_publish_date)
-    end
-
-    # This is a flakey spec. Due to how we are localizing date times, sometimes
-    # the date is off. See https://github.com/thepracticaldev/dev.to/issues/7086.
-    xit "renders the proper modified at date" do
-      article.update(edited_at: Time.zone.now)
-      get article.path
-      expect(response.body).to include GI.escapeHTML(article.edited_at.strftime("%b %d, %Y"))
-    end
-
-    it "renders the proper author" do
-      expect(response.body).to include CGI.escapeHTML(article.cached_user_username)
+    # rubocop:disable RSpec/ExampleLength
+    it "renders the proper JSON-LD for an article" do
+      expect(response_json).to include(
+        "@context" => "http://schema.org",
+        "@type" => "Article",
+        "mainEntityOfPage" => {
+          "@type" => "WebPage",
+          "@id" => URL.article(article)
+        },
+        "url" => URL.article(article),
+        "image" => [
+          ApplicationController.helpers.article_social_image_url(article, width: 1080, height: 1080),
+          ApplicationController.helpers.article_social_image_url(article, width: 1280, height: 720),
+          ApplicationController.helpers.article_social_image_url(article, width: 1600, height: 900),
+        ],
+        "publisher" => {
+          "@context" => "http://schema.org",
+          "@type" => "Organization",
+          "name" => "#{ApplicationConfig['COMMUNITY_NAME']} Community",
+          "logo" => {
+            "@context" => "http://schema.org",
+            "@type" => "ImageObject",
+            "url" => ApplicationController.helpers.cloudinary(SiteConfig.logo_png, 192, "png"),
+            "width" => "192",
+            "height" => "192"
+          }
+        },
+        "headline" => article.title,
+        "author" => {
+          "@context" => "http://schema.org",
+          "@type" => "Person",
+          "url" => URL.user(user),
+          "name" => user.name
+        },
+        "datePublished" => article.published_timestamp,
+        "dateModified" => article.edited_at&.iso8601 || article.published_timestamp,
```

Comment:
Instead of using a `||` here, I think we should be more decisive on what we expect `response_json` to have

Reply:
That's fair! The reason why I left the `||` here is because we've been running into issues with testing the `edited_at` date--this spec was flaky prior to this refactor because of the way we localize date times, causing the date to be off sometimes.

That being said though, I can remove the `edited_at` date in favor of the `published_timestamp`.